conda/environment.yml

```diff
@@ -3,27 +3,164 @@
+  # some more information
+  # - gh-scoped-creds==4.1
+  # More git tools from github-cli
+  - gh~=2.32.1
+
+  # VS Code support
+  # - jupyter-vscode-proxy~=0.5
+  # - code-server~=4.16.1
+
+  # Jupyter tools
+  - ipython~=8.20.0
+  - ipywidgets~=8.1.1
+  - jupyter-ai~=2.20
+  - jupyter-book~=1.0.0
+#  - jupyter-offlinenotebook~=0.2.2 # Outdated according to jupyterlab
+#  - jupyter-collaboration~=2.0.1
+  - jupyterlab-favorites~=3.2.1
+  - jupyterlab-geojson~=3.4.0
+  - jupyterlab-git~=0.50.0
+  - jupyterlab-h5web~=11.1.0
+  - jupyterlab-myst~=2.4.2
+#  - jupyterlab-plotly~=5.18.0 # Outdated accoring to jupyterlab
+  - jupyterlab_pygments~=0.3.0 # To bring extension uptodate
+  - jupytext~=1.16.1
+  - nbdime~=4.0.1
+  # JupyterBook Addons
   - sphinxcontrib-bibtex
-  # For building the Splashpage
-  - cookiecutter
-  # Dependencies of jinja-markdown:
-  - jinja2>=2.11
-  - pygments>=2.6.1
-  - pymdown-extensions>=7.1
-  - pip:
-    - jinja-markdown==1.210911
+  # dashboards
+  - voila~=0.5
+
+  # Core scientific python stack
+  - cython~=3.0.10 # optimization, C API access
+  - flox~=0.9.7 # optimization, xarray operations
+  - hypothesis==6.58.0 # needed by numpy testing tools
+  - networkx==2.8.8
+  - numba~=0.60.0 # high-performance numerics
+  - numpy==1.26.4
+  - pandas>=2.0.1,<3.0
+  - scikit-image~=0.24.0
+  - scikit-learn>=1.2.2,<2.0.0
+  - scipy>=1.10
+  - statsmodels==0.13.5
+  - sympy~=1.12
+  - xarray>=2024.05.0
+
+  # Visualization packages
+  - bokeh~=3.2.2
+  - cartopy~=0.23.0 # geospatial plotting with matplotlib
+  - geemap~=0.32
+  - geoviews~=1.10.1
+  - hvplot~=0.10.0
+  - ipyleaflet~=0.19.2
+  - ipympl~=0.9.3 # This enables matplotlib interaction with jupyter widgets
+  - leafmap~=0.36
+  - lonboard~=0.9
+  - matplotlib~=3.9.0
+  - plotly~=5.18.0
+  - seaborn~=0.13.2 # statistical plotting with matplotlib
+
+  # Geospatial data packages
+  - geopandas>=0.14.4
+  - pygmt~=0.12.0
+  - rasterio~=1.3
+  - rioxarray~=0.17.0
+
+  # File formats and file management, download, dataset caching
+  - h5py~=3.11
+  - h5netcdf~=1.3.0
+  - nco~=5.1.6
+  - pooch~=1.8.0
+  - zarr~=2.18
+  - kerchunk~=0.2
+
+  # Cloud access tools and libraries
+  - awscli~=1.27.76
+  - boto3
+  - s3fs>=2023.6.0
+  # handle login and similar details for accessing earthdata protected data
+  # https://github.com/nsidc/earthaccess
+  - earthaccess~=0.10.0
+  # Access datasets exposed via STAC
+  - pystac-client~=0.8.3
+  # Access datasets exposed via intake catalogs
+  - intake==0.6.6
+  - intake-esm>=2023.7.7
+  - intake-stac==0.4.0
+  - intake-xarray==0.6.1
+  - gcsfs>=2023.5.0
+
+  # Distributed computing
+  - dask>=2023.12.1
+  - dask-labextension~=7.0.0
+  - dask-geopandas~=0.4.1
+
+  # Other useful generic python packages
+  - pillow~=10.2.0 # Python imaging library, useful for many image-related tasks
+  - pytest~=8.3
+  - pytest-cov~=5.0
+  - pep8~=1.7
+  - flake8~=7.1
+  - tqdm~=4.66 # progress bars, with notebook support
```

Comment:
While I am fan of clean and tested code, probably overkill for building this book (Lines 127 - 132)

Reply:
Agreed, but I see `import tqdm` fairly often in notebooks so will keep that in. We also have a packaging tutorial which may discuss some of these things